src/partners/swapuz.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  asArray,
+  asNumber,
+  asObject,
+  asOptional,
+  asString,
+  asUnknown
+} from 'cleaners'
+import fetch from 'node-fetch'
+
+import { PartnerPlugin, PluginParams, PluginResult, StandardTx } from '../types'
+import { datelog } from '../util'
+
+const asSwapuzLogin = asObject({
+  result: asObject({
+    token: asString
+  })
+})
+
+const asSwapuzTx = asObject({
+  createDate: asString,
+  from: asString,
+  to: asString,
+  amountResult: asNumber,
+  amount: asNumber,
+  amountBTC: asNumber,
+  startAmount: asNumber
+})
+
+const asSwapuzRawTx = asObject({
+  id: asNumber,
+  status: asNumber,
+  wTxId: asOptional(asString),
+  dTxId: asOptional(asString)
+})
+
+const asSwapuzResult = asObject({
+  result: asObject({
+    currentPage: asNumber,
+    maxPage: asNumber,
+    result: asArray(asUnknown)
```

Comment:
If only we had lazy cleaners:

```ts
result: asArray(asLazy(asSwapuzRawTx))
```

Then we can look see how the cleaners are coupled at the definition level without needing to read the implementation.